ugs-platform/ugs-platform-plugin-joystick/src/main/java/com/willwinder/ugs/nbp/joystick/model/JoystickControl.java

```diff
@@ -0,0 +1,84 @@
+/*
+    Copyright 2020 Will Winder
+
+    This file is part of Universal Gcode Sender (UGS).
+
+    UGS is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    UGS is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with UGS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.willwinder.ugs.nbp.joystick.model;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The available buttons and axises that can be read by the service.
+ * The controls are divided into two groups:
+ *  - Analog - which will read analog values between 1.0 to -1.0
+ *  - Digital - which will read digital values such as buttons true/false
+ *
+ * @author Joacim Breiler
+ */
+public enum JoystickControl {
+    A("platform.plugin.joystick.a", false),
+    B("platform.plugin.joystick.b", false),
+    X("platform.plugin.joystick.x", false),
+    Y("platform.plugin.joystick.y", false),
+    BACK("platform.plugin.joystick.back", false),
+    GUIDE("platform.plugin.joystick.guide", false),
```

Comment:
I've never heard of the guide button, is it the same as select? After some searching, it looks like this is an xbox thing?

It's probably safe to borrow the terminology used by retroarch. I think the changes are:
GUIDE -> SELECT
LEFT_BUMPER -> L
LEFT_TRIGGER -> L2
LEFT_STICK -> L3
RIGHT_BUMPER -> R
RIGHT_TRIGGER -> R2
RIGHT_STICK -> L3

https://retropie.org.uk/docs/RetroArch-Configuration/#example-default-per-system-retroarchcfg
https://github.com/libretro/RetroArch/blob/a96ae6e987671da71be4aa840e80eb341d4d62f3/retroarch.cfg#L491
```
# input_player1_a_btn =
# input_player1_b_btn =
# input_player1_y_btn =
# input_player1_x_btn =
# input_player1_start_btn =
# input_player1_select_btn =
# input_player1_l_btn =
# input_player1_r_btn =
# input_player1_left_btn =
# input_player1_right_btn =
# input_player1_up_btn =
# input_player1_down_btn =
# input_player1_l2_btn =
# input_player1_r2_btn =
# input_player1_l3_btn =
# input_player1_r3_btn =
```

Reply:
Thanks, I'll update these!